pkg/fleet/internal/oci/download.go

```diff
@@ -0,0 +1,236 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package oci provides a way to interact with Datadog Packages OCIs.
+package oci
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"runtime"
+	"strings"
+
+	"github.com/awslabs/amazon-ecr-credential-helper/ecr-login"
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/name"
+	oci "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/empty"
+	"github.com/google/go-containerregistry/pkg/v1/google"
+	"github.com/google/go-containerregistry/pkg/v1/layout"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"github.com/google/go-containerregistry/pkg/v1/types"
+	httptrace "gopkg.in/DataDog/dd-trace-go.v1/contrib/net/http"
+
+	"github.com/DataDog/datadog-agent/pkg/fleet/internal/tar"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+const (
+	// RegistryAuthDefault is the default registry authentication method. Under the hood, it uses the Docker configuration.
+	RegistryAuthDefault string = "docker"
+	// RegistryAuthGCR is the Google Container Registry authentication method.
+	RegistryAuthGCR string = "gcr"
+	// RegistryAuthECR is the Amazon Elastic Container Registry authentication method.
+	RegistryAuthECR string = "ecr"
+)
+
+const (
+	// AnnotationPackage is the annotiation used to identify the package name.
+	AnnotationPackage = "com.datadoghq.package.name"
+	// AnnotationVersion is the annotiation used to identify the package version.
+	AnnotationVersion = "com.datadoghq.package.version"
+
+	// DatadogPackageLayerMediaType is the media type for the main Datadog Package layer.
+	DatadogPackageLayerMediaType types.MediaType = "application/vnd.datadog.package.layer.v1.tar+zstd"
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 Thought:** ‏To keep backwards compatibility this should never change, maybe we should add a comment

Reply:
I think we're good, it's also defined in datadog-packages and basically everything breaks if you don't have it, not just for the bootstrap but also for regular installs.